traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/http/HTTPAccessEventBuilder.java

```diff
@@ -84,7 +92,11 @@ public static String create(final HTTPAccessRecord httpAccessRecord) {
         String chi = formatObject(httpServletRequest.getRemoteAddr());
         final String url = formatRequest(httpServletRequest);
         final String cqhm = formatObject(httpServletRequest.getMethod());
-        final String cqhv = formatObject(httpServletRequest.getProtocol());
+        String cqhv = "-";
+        final String protocol = formatObject(httpServletRequest.getProtocol());
+        if (validProtocol(protocol)) {
+            cqhv = protocol;
+        }
```

Comment:
`formatObject()` already defaults the string to `-` if the argument is null. Rather than making a new `validProtocol()` function, should `formatObject()` check for empty strings too?